enterprise/metrics/src/main/java/org/neo4j/metrics/MetricsExtension.java

```diff
@@ -34,101 +29,79 @@
-import org.neo4j.metrics.output.GangliaOutput;
-import org.neo4j.metrics.output.GraphiteOutput;
-import org.neo4j.metrics.source.Neo4jMetricsFactory;
+import org.neo4j.metrics.output.EventReporter;
+import org.neo4j.metrics.output.OutputBuilder;
+import org.neo4j.metrics.source.Neo4jMetricsBuilder;
 
 public class MetricsExtension implements Lifecycle
 {
-    private final LifeSupport life;
+    private final LifeSupport life = new LifeSupport();
+    private final MetricsKernelExtensionFactory.Dependencies dependencies;
     private final LogService logService;
     private final Config configuration;
-    private final Monitors monitors;
-    private final TransactionCounters transactionCounters;
-    private final PageCacheMonitor pageCacheCounters;
-    private final CheckPointerMonitor checkPointerMonitor;
-    private final IdGeneratorFactory idGeneratorFactory;
-    private final LogRotationMonitor logRotationMonitor;
-    private final DataSourceManager dataSourceManager;
-    private final DependencyResolver dependencyResolver;
     private final KernelContext kernelContext;
 
+
     public MetricsExtension( MetricsKernelExtensionFactory.Dependencies dependencies )
     {
-        life = new LifeSupport();
-        logService = dependencies.logService();
-        configuration = dependencies.configuration();
-        monitors = dependencies.monitors();
-        dataSourceManager = dependencies.dataSourceManager();
-        transactionCounters = dependencies.transactionCounters();
-        pageCacheCounters = dependencies.pageCacheCounters();
-        checkPointerMonitor = dependencies.checkPointerCounters();
-        logRotationMonitor = dependencies.logRotationCounters();
-        idGeneratorFactory = dependencies.idGeneratorFactory();
-        dependencyResolver = dependencies.getDependencyResolver();
-        kernelContext = dependencies.kernelContext();
+        this.dependencies = dependencies;
+        this.logService = dependencies.logService();
+        this.configuration = dependencies.configuration();
+        this.kernelContext = dependencies.kernelContext();
     }
 
     @Override
-    public void init() throws Throwable
+    public void init()
     {
         Log logger = logService.getUserLog( getClass() );
+        logger.info( "Initiating metrics..." );
 
         // Setup metrics
         final MetricRegistry registry = new MetricRegistry();
 
-        logger.info( "Initiating metrics.." );
-
         // Setup output
-        String prefix = computePrefix( configuration );
-
-        life.add( new CsvOutput( configuration, registry, logger, kernelContext ) );
-        life.add( new GraphiteOutput( configuration, registry, logger, prefix ) );
-        life.add( new GangliaOutput( configuration, registry, logger, prefix ) );
+        EventReporter reporter = new OutputBuilder( configuration, registry, logger, kernelContext, life ).build();
+        boolean outputBuilt = reporter != null;
```

Comment:
instead of doing this check and return null as result of build() method call, we can simplify this kind of case by always returning EventReporter and use isEmpty() method which is already defined there to check do we need to report anything